handler.go

```diff
@@ -194,5 +194,25 @@ func (pset pathConfigSet) find(path string) (pc *pathConfig, subpath string) {
 	if i > 0 && strings.HasPrefix(path, pset[i-1].path+"/") {
 		return &pset[i-1], path[len(pset[i-1].path)+1:]
 	}
-	return nil, ""
+
+	// Otherwise now look for the shortest prefix pathConfig
+	// For example:
```

Comment:
Instead of drilling in commentary on this part, I would instead comment on the binary search above as being the "fast path" and documenting where it fails (the test case I came up with was ["/", "/abc", "/xyz"] and "/def".